Scoping doc

yulric opened the pull request.

yulric committed:
This commit exists to preserve Doug's work on the scoping document.

DougManuel commented:
1) Wide versus long may be better situated as a future feature.
2) Do we want to include concepts such as logging and testing? Having clear expectations for the user about what will happen during transformations and the ability to debug, examine transformation effects, etc. Again, this could go in the next document.

DougManuel reviewed (dismissed):
I've made a few comments and suggestions. 

We can close the PR whenever you want and address any outstanding issues with the specifications. 

However, feel free to ask for more clarity.

yulric commented:
> 1. Wide versus long may be better situated as a future feature.
> 2. Do we want to include concepts such as logging and testing? Having clear expectations for the user about what will happen during transformations and the ability to debug, examine transformation effects, etc. Again, this could go in the next document.

1. Agreed
2. Makes sense for those to go into the next document since its hard to provide a high level overview of what should go into logging and testing without getting into the specifications.

yulric committed:
added scoping document

yulric commented:
@DougManuel The scoping document should now be ready for you to review. The document is in the scope-docs folder and can be rendered using the command `quarto preview ./scope-docs`. You will need to delete and re-pull your version of this branch since I changed the entire history. I added a commit with everything you wrote to ensure that its all preserved. I also pulled in the changed from your catalog branch into the scope. Finally, I created a v2 branch into which this branch will be merged.

DougManuel commented:
Small point: the current version is 0.1.0. Let's call this update v1.0.0 (not v2).

DougManuel commented:
@yulric I didn't see the catalog.qmd in the commit list, but we can access the document from this closed PR https://github.com/Big-Life-Lab/recodeflow/pull/65.

yulric commented:
@DougManuel Let me know once you're done with your review

DougManuel commented:
I've finished with my reviews.
